types/index.d.ts

```diff
@@ -238,7 +238,10 @@ export namespace Service {
     objectMode: boolean;
   }
 
-  type TransportFunction = () => void; // TODO
+  
+
+  type TransportFunctionCallback = (err: Error|null|undefined, res?: stream.Stream) => stream.Stream;
+  type TransportFunction = (cb: TransportFunctionCallback) => void; // TODO
```

Comment:
I think the return types are swapped. The outer function is the one which is expected to return a stream:

```typescript
type TransportFunctionCallback = (err: Error|null|undefined, res?: stream.Stream) => void;
type TransportFunction = (cb: TransportFunctionCallback) => stream.Stream;
```

Reply:
Doh! You're right, that makes way more sense. I've pushed a fix.